scripts/windows_dev_setup.ps1

```diff
@@ -0,0 +1,352 @@
+            while ((Get-Item "protoc-$global:protoc_version-win64.zip").Length -lt 2MB) {
+                Start-Sleep -Seconds 1
+            }
+            Expand-Archive -Path "protoc-$global:protoc_version-win64.zip" -DestinationPath "$env:USERPROFILE\protoc-$global:protoc_version-win64" -ErrorAction SilentlyContinue
+        }
+        else {
+            # Download and extract the 32-bit version of Protoc
+            Invoke-WebRequest -Uri "https://github.com/protocolbuffers/protobuf/releases/download/v$global:protoc_version/protoc-$global:protoc_version-win32.zip" -OutFile "protoc-$global:protoc_version-win32.zip" -ErrorAction SilentlyContinue
+            while ((Get-Item "protoc-$global:protoc_version-win32.zip").Length -lt 2MB) {
+                Start-Sleep -Seconds 1
+            }
+            Expand-Archive -Path "protoc-$global:protoc_version-win32.zip" -DestinationPath "$env:USERPROFILE\protoc-$global:protoc_version-win32" -ErrorAction SilentlyContinue
+        }
+        
+        # Add the Protoc installation directory to the user's PATH environment variable
+        [Environment]::SetEnvironmentVariable("PATH", "$env:PATH;$env:USERPROFILE\protoc-$global:protoc_version-win64\bin", "User")
+    }
+}
+
+function install_cargo_plugins {  # Installs Grcov, protoc components, and cargo components
+    cargo install protoc-gen-prost --locked
+    cargo install protoc-gen-prost-serde --locked
+    cargo install protoc-gen-prost-crate --locked
+    cargo install grcov --version $global:grcov_version --locked
+    cargo install cargo-sort --locked
+    cargo install cargo-nextest --locked
+}
+
+function install_llvm {
+  $result = check_package "LLVM"
+  if ($result) {
+	winget install LLVM.LLVM --silent
+	}
+  else {
+	winget upgrade --id LLVM.LLVM
+  }
+}
+
+function install_openssl {
+  $result = check_package "OpenSSL"
+  if ($result) {
+	winget install ShiningLight.OpenSSL --silent
+	}
+  else {
+	winget upgrade --id ShiningLight.OpenSSL --silent
+  }
+}
+
+function install_nodejs {
+  $result = check_package "Node.js"
+  if ($result) {
+	winget install OpenJS.NodeJS --silent
+	}
+  else {
+	winget upgrade --id OpenJS.NodeJS -silent
+  }
+}
+
+function install_python {
+  $result = check_package "Python"
+  if ($result) {
+	winget install Python.Python.3.11 --silent
+	}
+  else {
+	winget upgrade --id Python.Python.3.11 -silent
+  }
+  python -m pip install --upgrade pip
+  python -m pip install schemathesis
+}
+
+function install_pnpm {
+  $result = check_package "pnpm"
+  if ($result) {
+	winget install pnpm.pnpm --silent
+	}
+  else {
+    winget upgrade --id LLVM.LLVM
+  }
+}
+
+function install_postgresql {
+  $result = check_package "PostgreSQL"
+  if ($result) {
+	winget install PostgreSQL.PostgreSQL --silent
+	}
+  else {
+	winget upgrade --id LLVM.LLVM
+   }
+}

```

Comment:
Postgres isn't actually required, we'll probably want to follow up later on removing this as a requirement, but for V1 it's fine.